src/Files.App/ViewModels/Settings/GeneralViewModel.cs

```diff
@@ -152,6 +152,11 @@ private void AddSupportedAppLanguages()
 				.ThenBy(language => language.LanguageName);
 			AppLanguages = new ObservableCollection<AppLanguageItem>(appLanguages);
 
+			// Add default language
+			var osDefaultLanguage = new AppLanguageItem(CultureInfo.InstalledUICulture.Name);
+			if (AppLanguages.Select(language => language.LanguageName.Contains(osDefaultLanguage.LanguageName)).Any())
+				AppLanguages[0].LanguagID = osDefaultLanguage.LanguagID;
```

Comment:
I don't think we need to make any changes here, it should be possible to handle everything from the format extension class.

Reply:
The reason for the change to GeneralViewModel.cs was to store the language name, thus avoiding string.Empty, but this is also addressed in a future helper class